tuf/client_rework/updater_rework.py

```diff
@@ -142,46 +143,38 @@ def updated_targets(targets: Dict, destination_directory: str) -> Dict:
 
         return updated_targets
 
-    def download_target(self, target: Dict, destination_directory: str):
+    def download_target(
+        self,
+        targetinfo: Dict,
+        destination_directory: str,
+        target_base_url: Optional[str] = None,
```

Comment:
Following on from above, I think this makes sense as an argument to replace `repository_base_url` + `default_targets_dir` – i.e. in the case where targets files are on an entirely different host.